src/Microsoft.ComponentDetection.Detectors/pip/PipReportComponentDetector.cs

```diff
@@ -75,6 +84,26 @@ protected override async Task<IObservable<ProcessRequest>> OnPrepareDetectionAsy
             return Enumerable.Empty<ProcessRequest>().ToObservable();
         }
 
+        this.CurrentScanRequest.DetectorArgs.TryGetValue("Pip.PythonExePath", out var pythonExePath);
+        if (!await this.pythonCommandService.PythonExistsAsync(pythonExePath))
+        {
+            this.Logger.LogInformation($"No python found on system. Python detection will not run.");
+
+            return Enumerable.Empty<ProcessRequest>().ToObservable();
+        }
+        else
+        {
+            var pythonVersion = await this.pythonCommandService.GetPythonVersionAsync(pythonExePath);
+            this.pythonResolver.SetPythonEnvironmentVariable("python_version", pythonVersion);
+
+            var pythonPlatformString = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
```

Comment:
How are we resolving MacOS?

Reply:
updating to use actual python script for this: https://docs.python.org/3/library/sys.html#sys.platform